### PR TITLE
more interfaces in typescript typings

### LINKED ISSUE #152: invalid syntax in 1.0 typings
### Reproduction demo

(codesandbox doesn't show errors in definition files. easily reproducible in VS Code or with `tsc` directly.)

### Steps to reproduce the problem

1.  import `react-popper @ 1.0.0-beta.5` in a typescript project
2.  witness compiler error

### What is the expected behavior?

success!

### What went wrong?

https://github.com/FezVrasta/react-popper/blob/master/typings/react-popper.d.ts#L24 

produces the following error in TypeScript 2.7.2:

```
JSDoc types can only be used inside documentation comments.
```

### Any other comments?

the syntax looks invalid. `placement: ?PopperJS.Placement` shouldn't have a leading `?`, but i'm not sure if `placement?` is correct or if the `placement` property will always be defined?

### Packages versions

- Popper.js: 1.14.1
- react-popper: 1.0.0-beta.5

### PR DESCRIPTION
**Pull request by giladgray**
#### fixes #152 

split typescript types into more interfaces that can be used separately in app code

```tsx
import * as React from "react";
import { Manager, Popper, PopperChildrenProps, Reference, RefProps } from "react-popper";

export class Component extends React.Component {
    public render() {
        return <Manager><Popper>{this.renderPopper}</Popper></Manager>;
    }
    private renderPopper = (props: PopperChildrenProps) => {
        // profit! reuse the same function instance!
    }
}

### CODE REVIEW

**Comment by FezVrasta** [comment]
Thanks, is there any way to keep `RefProps` private? (not exported)

File: typings/react-popper.d.ts [hunk 1]
```diff
@@ -2,32 +2,36 @@ import * as React from "react";
...
+
 interface ReferenceProps {
-  children: (props: ({
-    ref: (ref: HTMLElement | null) => void,
-  })) => React.ReactNode;
+    children: (props: RefProps) => React.ReactNode;
 }
 export class Reference extends React.Component<ReferenceProps, {}> { }
 
+interface ArrowProps extends RefProps {
```
Comment (FezVrasta) [inline typings/react-popper.d.ts]: `PopperArrowProps` instead? So we keep all the Popper related types under the same namespace?

**Comment by giladgray** [comment]
@FezVrasta not possible to prevent symbols in the .d.ts file from being exported, but i replaced `RefProps` with a simple `RefHandler` type.

File: typings/react-popper.d.ts [hunk 2]
```diff
@@ -6,28 +6,36 @@ interface ManagerProps {
...
 
+interface PopperArrowProps {
+  ref: RefHandler;
+  style: React.CSSProperties;
+}
+
+interface PopperChildrenProps {
+  arrowProps: PopperArrowProps;
+  outOfBoundaries: boolean | null;
+  placement?: PopperJS.Placement;
```
Comment (giladgray) [inline typings/react-popper.d.ts]: @FezVrasta do we know if this field will always be defined? as in, should it be marked optional or not? my suspicion is that it's always defined, that popper.js always computes a valid `placement` regardless of the given option value

Comment (FezVrasta) [reply typings/react-popper.d.ts]: Yes actually I think it's always going to be defined, I'm not sure why I marked it as optional.

**Pull request merged**